frontend/src/utils/__tests__/helper.test.js

```diff
@@ -0,0 +1,23 @@
+import { describe, it, expect } from "vitest";
+
+import { formatDate } from "../helper";
+
+describe("formatDate", () => {
+  it("month and day strings each are length 2", () => {
+    expect(formatDate("1000-10-10")).toBe("1000-10-10");
+  });
+  it("month string is length 1", () => {
+    expect(formatDate("1000-9-10")).toBe("1000-09-10");
+  });
+  it("day string is length 1", () => {
+    expect(formatDate("1000-10-9")).toBe("1000-10-09");
+  });
+  it("year string is length 2", () => {
+    expect(formatDate("01-10-09")).toBe("2009-01-10");
```

Comment:
Does "01/10/09" work? Or "1000/09/10"? They *should*, and since you've tried various combinations with dashes it probably makes sense to try. With and without padding? It's not a complicated function, so it's not too hard to be reasonably exhaustive.

Reply:
I think this is a good idea for a test case in a component or end-to-end integration or functional test as a part of our contract to the user. However, since the Javascript `Date()` constructor is responsible for parsing the input date string, I do not think it is appropriate for our unit test to reproduce tests that likely already exist for `Date()`.

Reply:
The argument isn't entirely invalid; however if you really intending to follow that philosophy then you've already got more test cases than necessary. By that logic I would expect to see one happy path and one error path, not the multiple "redundant" happy paths you've got.

Reply:
ah, do you mean like an input of `1000-9-9`?

Reply:
If you're implicitly trusting `new Date`'s parser, I don't think it really matters what string you use -- one string that is correctly parsed and one that isn't should be fine. That the "correct" one isn't trivially straightforward (e.g., your non-zero-padded example) is probably a good idea, but, again, if there's any actual question whether any of the possible expected variants will work we're *not* implicitly trusting `Date`.

Reply:
oh! I forgot one of the test cases was for regression, so that we don't reintroduce a potential off by one error

Reply:
should I keep that?